### PR TITLE
🐛 Source Asana: migrate to advancedAuth

### LINKED ISSUE #26225: Source Asana: Migrate to advancedAuth

### PR DESCRIPTION
**Pull request by arsenlosenko**
## What
Resolving this issue:
https://github.com/airbytehq/airbyte/issues/26225

## How
Remove `authSpecification` from specification, use `advancedAuth` instead

### CODE REVIEW

File: airbyte-integrations/connectors/source-asana/source_asana/spec.json [hunk 1]
```diff
@@ -65,12 +65,48 @@
       }
     }
   },
-  "authSpecification": {
-    "auth_type": "oauth2.0",
-    "oauth2Specification": {
-      "rootObject": ["credentials", 1],
-      "oauthFlowInitParameters": [["client_id"], ["client_secret"]],
-      "oauthFlowOutputParameters": [["refresh_token"]]
+  "advanced_auth": {
```
Comment (davydov-d) [inline airbyte-integrations/connectors/source-asana/source_asana/spec.json:68]: you need to add predicate key and predicate value

**Review by davydov-d — CHANGES_REQUESTED** [review]

**Review by bazarnov — APPROVED** [review]

File: airbyte-integrations/connectors/source-asana/source_asana/spec.json (hunk 1 above)
Comment (lazebnyi) [reply airbyte-integrations/connectors/source-asana/source_asana/spec.json:68]: @arsenlosenko did you check it manually?

**Review by lazebnyi — CHANGES_REQUESTED** [review]

File: airbyte-integrations/connectors/source-asana/source_asana/spec.json (hunk 1 above)
Comment (arsenlosenko) [reply airbyte-integrations/connectors/source-asana/source_asana/spec.json:68]: not yet, checking it right now, I'll request review again once all things are checked and my changes are updated

**Review by davydov-d — APPROVED** [review]

**Review by lazebnyi — APPROVED** [review]

**Pull request merged**